src/pygeoprocessing/routing/routing.pyx

```diff
@@ -3354,16 +3355,17 @@ def extract_strahler_streams_d8(
                 is_drain = 0
                 x_l = xoff + i
                 y_l = yoff + j
-                local_flow_accum = <long>flow_accum_managed_raster.get(
+                local_flow_accum = <double>flow_accum_managed_raster.get(
                     x_l, y_l)
-                if local_flow_accum < min_flow_accum_threshold:
+                if (local_flow_accum < min_flow_accum_threshold or
+                        _is_close(local_flow_accum,
+                                  flow_accum_nodata, 1e-8, 1e-5)):
                     continue
                 # check to see if it's a drain
                 d_n = <int>flow_dir_managed_raster.get(x_l, y_l)
                 x_n = x_l + D8_XOFFSET[d_n]
                 y_n = y_l + D8_YOFFSET[d_n]
-
-                if (x_n < 0 or y_n < 0 or x_n >= n_cols or y_n >= n_cols or
+                if (x_n < 0 or y_n < 0 or x_n >= n_cols or y_n >= n_rows or

```

Comment:
This was an unrelated bug, `y_n >= n_rows` is the fix. A similar subsequent check for this meant that this bug never resulted in any bad results, only non-optimal searches. Given that, I didn't see a reasonable way to test this.

Reply:
Sounds ok to me!  There are a number of things that have come up in pygeoprocessing's routing over the years that are subtle bugs that are difficult to test or required impractically large test data.  In both these cases, we just patched them and didn't worry about the tests.  I'm a-ok with continuing this practice if you are!